### DISCUSSION TITLE
Using operators inside embeddables is not allowed

### DISCUSSION THREAD

**Discussion by dungkaka** [mikro-orm/mikro-orm#4930] [General]
I have a entity data:
```ts
class Test {
    @Embedded(() => Author, { array: true, nullable: true })
    authors: Author[] = [];
}

@Embeddable()
class Author {
    @Property({ nullable: true })
    id: number;

    @Property()
    name: string;
}

const qb = this.em.createQueryBuilder(Test).select("*").where({
    authors: { $contains: [{ id: 1 }] }
})

```
It throw error: ValidationError: Using operators inside embeddables is not allowed, move the operator above.
That I want is run query sql like:
```sql
SELECT * FROM test WHERE authors @> '[{"id": 1}]'
```

How can I deal with this?

MikroORM: 5.9.2
NodeJS: 16

**Comment by B4nan** [comment]
This should help (you will need to update to v6 to have it)

https://github.com/mikro-orm/mikro-orm/commit/ecf1f0c96484178a4d7fc8228c2a41198d20ec30

**Reply by dungkaka** [reply]
Thanks, I will update to v6 and test it !
